### ISSUE TITLE
`Parameter` range is incorrect for `*args` and `**kwargs`

### ISSUE THREAD

**Issue by GtrMo** [astral-sh/ruff#10282]
I'm working on a fix for the `flake8-unused-arguments` (ARG) rules that would delete them.

While working on this, I noticed that the `*args` and `**kwargs` `Parameter.range` does not include the `*`. 
I think that `Parameter.range` should include `*` or `**`, leaving `Parameter.name.range` for the actual name range if needed.

```python
def f(*args, **kwargs): pass
#      ~~~~    ~~~~~~    <-- actual range
#     ^^^^^  ^^^^^^^^    <-- expected range
```

<details>
<summary>See the currently generated AST</summary>

https://github.com/astral-sh/ruff/blob/7b4a73d421c130430883bb944a285f096e3aa9a5/crates/ruff_python_parser/src/snapshots/ruff_python_parser__function__tests__function_pos_and_kw_only_args_with_defaults_and_varargs_and_kwargs.snap#L58-L67

Here I would expect the `Parameter` range to be 1 longer than the `Identifier` range.

https://github.com/astral-sh/ruff/blob/7b4a73d421c130430883bb944a285f096e3aa9a5/crates/ruff_python_parser/src/snapshots/ruff_python_parser__function__tests__function_pos_and_kw_only_args_with_defaults_and_varargs_and_kwargs.snap#L124-L134

And here I would expect the `Parameter` range to be 2 longer than the `Identifier` range.
</details>

If this is indeed considered a bug, I have a PR ready to fix it.
